src/vehicle.cpp

```diff
@@ -1057,28 +1058,47 @@ void CallVehicleTicks()
 		CommandCost res = Command<CMD_AUTOREPLACE_VEHICLE>::Do(DC_EXEC, v->index);
 		SubtractMoneyFromCompany(CommandCost(EXPENSES_NEW_VEHICLES, -(Money)c->settings.engine_renew_money));
 
-		if (!IsLocalCompany()) continue;
+		if (IsLocalCompany()) {
+			if (res.Succeeded()) {
+				ShowCostOrIncomeAnimation(x, y, z, res.GetCost());
+			}else {
+				StringID error_message = res.GetErrorMessage();
+				if (error_message != STR_ERROR_AUTOREPLACE_NOTHING_TO_DO && error_message != INVALID_STRING_ID) {
+					if (error_message == STR_ERROR_NOT_ENOUGH_CASH_REQUIRES_CURRENCY)
+						error_message = STR_ERROR_AUTOREPLACE_MONEY_LIMIT;
+
+					StringID message;
+					if (error_message == STR_ERROR_TRAIN_TOO_LONG_AFTER_REPLACEMENT) {
+						message = error_message;
+					} else {
+						message = STR_NEWS_VEHICLE_AUTORENEW_FAILED;
+					}
 
-		if (res.Succeeded()) {
-			ShowCostOrIncomeAnimation(x, y, z, res.GetCost());
-			continue;
+					SetDParam(0, v->index);
+					SetDParam(1, error_message);
+					AddVehicleAdviceNewsItem(message, v->index);
+				}
+			}
 		}
 
-		StringID error_message = res.GetErrorMessage();
-		if (error_message == STR_ERROR_AUTOREPLACE_NOTHING_TO_DO || error_message == INVALID_STRING_ID) continue;
+		// Copy wagons from another train
+		if (v->type == VehicleType::VEH_TRAIN && v->copy_wagons_from != v->index){
```

Comment:
missing space (various other places too)
```suggestion
		if (v->type == VehicleType::VEH_TRAIN && v->copy_wagons_from != v->index) {
```